### ISSUE TITLE
False positive: Don't use function or method calls in WHERE clause

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#1174]
The following is stripped from an actual example:

````
BLOCK-LEVEL ON ERROR UNDO, THROW.

DEFINE BUFFER bu1 FOR stbrlay.
DEFINE BUFFER bu2 FOR stbrlayf.

DEFINE VARIABLE id AS INTEGER NO-UNDO.

FOR EACH bu1
   WHERE bu1.table_id  =  id
   OR    (
            CAN-FIND( FIRST bu2 WHERE bu2.table_id_report = id )
   )
NO-LOCK:
   // boo
END.
````
The following changes prevent the rule from flagging:

- remove the parentheses from the ``OR`` 
- replace variable ``id`` with a constant

![Image](https://github.com/user-attachments/assets/8b97120f-e42d-4395-a0b9-c2e4652a58ca)

**Comment by gquerret** [comment]
Will be improved.
